[tfjs-react-native] Build and setup issue with camera

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): `Ubuntu 20.04`
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: `Pixel_3a_API_30_x86.avd`
- TensorFlow.js installed from (npm or script link):
- TensorFlow.js version: `npm install @tensorflow/tfjs`, 
- CUDA/cuDNN version:


**Describe the problem**

I am following the steps to setup tfjs react native, ive gone from step 1 - step 4 however i am blocked with an error message after i run when i use either `npm run web` or `npm run android`. The problem both relate to the camera, i have installed the prequisite `expo install expo-camera`.


**Provide the exact sequence of commands / steps that you executed before running into the problem**

1. `npm run web`

```
Failed to compile.
/home/kay/kay/powerlevelai/mobileapp/node_modules/@tensorflow/tfjs-react-native/dist/camera/camera_stream.js 281:12
Module parse failed: Unexpected token (281:12)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|             const cameraComp = (
|             //@ts-ignore see https://github.com/microsoft/TypeScript/issues/30650
>             <CameraComponent key='camera-with-tensor-camera-view' {...(cameraProps)} ref={(ref) => (this.camera = ref)}/>);
|             // Create the glView if the camera has mounted.
|             let glViewComponent = null;

```

alternatively i have setup an android emulator using AVD.

1. `npm run android`

```
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.3/userguide/command_line_interface.html#sec:command_line_warnings

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:compileDebugJavaWithJavac'.
> Could not resolve all task dependencies for configuration ':app:debugCompileClasspath'.
   > Could not find com.google.android:cameraview:1.0.0.
     Required by:
         project :app > project :expo-camera
```


**Any other info / logs**

**package.json**

```
{
  "main": "index.js",
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "web": "expo start --web",
    "start": "react-native start",
    "test": "jest"
  },
  "dependencies": {
    "@react-native-async-storage/async-storage": "^1.14.1",
    "@react-native-community/async-storage": "^1.12.1",
    "@tensorflow/tfjs": "^3.3.0",
    "@tensorflow/tfjs-react-native": "^0.5.0",
    "expo": "~40.0.0",
    "expo-camera": "~9.1.0",
    "expo-gl": "~9.2.0",
    "expo-gl-cpp": "~9.2.0",
    "expo-splash-screen": "~0.8.0",
    "expo-updates": "~0.4.0",
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "react-native": "~0.63.4",
    "react-native-fs": "^2.16.6",
    "react-native-gesture-handler": "~1.8.0",
    "react-native-reanimated": "~1.13.0",
    "react-native-screens": "~2.15.0",
    "react-native-unimodules": "~0.12.0",
    "react-native-web": "~0.13.12"
  },
  "devDependencies": {
    "@babel/core": "~7.9.0",
    "@types/react": "~16.9.35",
    "@types/react-dom": "~16.9.8",
    "@types/react-native": "~0.63.2",
    "babel-preset-expo": "~8.3.0",
    "jest-expo": "~40.0.0",
    "typescript": "~4.0.0"
  },
  "jest": {
    "preset": "react-native"
  },
  "private": true
}
```
**App.tsx**

```
import * as tf from '@tensorflow/tfjs';
import '@tensorflow/tfjs-react-native';
import React, { Component } from 'react';
export default class App extends React.Component {
  constructor(props: any) {
    super(props);
    this.state = {
      isTfReady: false,
    };
  }

  async componentDidMount() {
    // Wait for tf to be ready.
    await tf.ready();
    // Signal to the app that tensorflow.js can now be used.
    this.setState({
      isTfReady: true,
    });
  }


  render() {
    return null;
  }
}
```

--- a/tfjs-react-native/README.md
+++ b/tfjs-react-native/README.md
@@ -102,9 +102,9 @@ If you use expo and encounter a build failure when running `npm run web` due to
 the steps below to make expo correctly transpile tfjs packages:
 
 - Run: `expo customize:web`
-  - Use the space key to select the `webpack.config.js` entry, then press "enter"
+  - Use the space key to select the `webpack.config.js` entry, then press "enter".
   - This will create a bare-minimum `webpack.config.js` file.
-- Edit the `webpack.config.js` file as follows
+- Edit the `webpack.config.js` file as follows:
 
 ```
 const createExpoWebpackConfigAsync = require('@expo/webpack-config');

--- a/tfjs-react-native/README.md
+++ b/tfjs-react-native/README.md
@@ -97,6 +97,35 @@ export class App extends React.Component {
 }
 ```
 
+If you use expo and encounter a build failure when running `npm run web` due to
+`You may need an appropriate loader to handle this file type...` error, follow
+the steps below to make expo correctly transpile tfjs packages:
+
+- Run: `expo customize:web`
+  - Use the space key to select the `webpack.config.js` entry, then press "enter"
+  - This will create a bare-minimum `webpack.config.js` file.
+- Edit the `webpack.config.js` file as follows
+
+```
+const createExpoWebpackConfigAsync = require('@expo/webpack-config');
+
+module.exports = async function(env, argv) {
+  const config = await createExpoWebpackConfigAsync(
+      {
+        ...env,
+        babel: {
+          dangerouslyAddModulePathsToTranspile: [
+            // Ensure that all packages starting with @tensorflow are
+            // transpiled.
+            '@tensorflow',
+          ],
+        },
+      },
+      argv);
+  return config;
+};
+```
+
 You can take a look at [`integration_rn59/App.tsx`](integration_rn59/App.tsx) for an example of what using tfjs-react-native looks like. In future we will add an example to the [tensorflow/tfjs-examples](https://github.com/tensorflow/tfjs-examples) repository.
 The [Webcam demo folder](integration_rn59/components/webcam) has an example of a style transfer app.
 

